src/c4/ofpstream.cc

```diff
@@ -28,9 +28,11 @@ using namespace std;
  * \param filename Name of the file to which synchronized output is to be
  * written.
  */
-ofpstream::ofpstream(std::string const &filename) : std::ostream(&sb_) {
+ofpstream::ofpstream(std::string const &filename, ios_base::openmode const mode)
```

Comment:
Please add `param` doxygen comment for `mode`.  Maybe mention that mode is an optional argument.